controllers/machinedeletionremediation_controller.go

```diff
@@ -90,62 +90,46 @@ func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, re
 		return ctrl.Result{}, err
 	}
 
-	annotations := remediation.GetAnnotations()
-	if annotations != nil {
-		machineNameNamespace, exists := annotations[MachineNameNamespaceAnnotation]
-		if exists {
-			machineName, machineNamespace, err := extractNameAndNamespace(machineNameNamespace, remediation.GetName())
-			if err != nil {
-				log.Error(err, "could not get Machine data from remediation", "remediation", remediation.GetName(), "annotation", machineNameNamespace)
-				return ctrl.Result{}, nil
-			}
+	log.Info("Machine Deletion Remediation CR found", "name", remediation.GetName())
 
-			ok, err := r.verifyMachineIsDeleted(ctx, machineName, machineNamespace, remediation.Name)
-			if err != nil {
-				return ctrl.Result{}, err
-			}
-			if !ok {
-				return ctrl.Result{RequeueAfter: 30 * time.Second}, nil
-			}
+	var machine *unstructured.Unstructured
+	if machine, err = r.getMachine(remediation); err != nil {
+		if apiErrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-	}
-
-	var node *v1.Node
-	if node, err = r.getNodeFromMdr(remediation); err != nil {
+		log.Error(err, "Unexpected error fetching Machine from Node", "node", remediation.Name)
 		return ctrl.Result{}, err
 	}
 
-	log.Info("MDR CR and affected Node found", "node", node.GetName())
+	log.Info("node-associated machine found", "node", remediation.Name, "machine", machine.GetName())
 
-	var machine *unstructured.Unstructured
-	if machine, err = r.buildMachineFromNode(node); err != nil {
-		r.Log.Error(err, noMachineFoundError, "node name", node.Name)
-		return ctrl.Result{}, err
-	}
-
-	if !hasControllerOwner(machine) {
-		log.Info("ignoring remediation of node-associated machine: the machine has no controller owner", "machine", machine.GetName(), "node name", remediation.Name)
-		return ctrl.Result{}, nil
-	}
-
-	log.Info("node-associated machine found", "node", node.Name, "machine", machine.GetName())
+	if !isMachineMarkedForDeletion(machine) {
```

Comment:
Nit:
In a very broad overview this is how I read the code (replacing if/else order):
```
if isMachineMarkedForDeletion {
 //Just do a requeue
} else{
  //1. Mark for deletion
  //2. Try to delete the machine
}
```
I think it's a bit confusing since I'd expect we will attempt to delete a Machine that is Marked for deletion.
Maybe instead of `isMachineMarkedForDeletion` something like `isMachineMarkedAsDeleted`  ?

Reply:
I started believing that I misunderstood your original suggestion for "markedForDeletion".
The code does exactly what you outlined. 
`isMarkedForDeletion` here is a poor wrapping of checking if `deletionTimeStamp` is set. What we do is (let's say Machine still exists)

```
if Machine has not yet deletionTimeStamp
  if Machine has owner
    save Machine's data to track its deletion phase
    delete Machine
    re-queue
else
  track machine deletion phase
  re-queue
```

the `save Machine's data` step is only for logging and provide feedback to the fact that the machine is actually deleted at some point, and also necessary, otherwise when Machine's `Node` disappear, we won't be able to directly check if the Machine's still there.

Reply:
Got it, it was my mistake :+1:

Reply:
well, if the code isn't clear, it's code's problem :wink: